### PR TITLE
Bug fix: id token Base64 decoding

### PR DESCRIPTION
**Pull request by lalimasharda**
Addresses base64 decoding errors reported by customers and #6492

### CODE REVIEW

File: lib/msal-browser/src/encode/Base64Decode.ts [hunk 1]
```diff
@@ -21,6 +21,19 @@ export function base64Decode(input: string): string {
...
+        case 0:
+            break;
+        case 2:
+            encodedString += "==";
+            break;
+        case 3:
+            encodedString += "=";
+            break;
+        default:
+            throw new Error("Invalid base64 string");
```
Comment (tnorling) [inline lib/msal-browser/src/encode/Base64Decode.ts]: Can we make this a first class error? e.g. a BrowserAuthError?

File: lib/msal-browser/test/encode/Base64Decode.spec.ts [hunk 2]
```diff
@@ -95,6 +95,25 @@ describe("Base64Decode.ts Unit Tests", () => {
             );
         });
 
+        it("Decode idtokenclaims with special characters", () => {
```
Comment (tnorling) [inline lib/msal-browser/test/encode/Base64Decode.spec.ts:99]: There's a bunch of [localization test cases](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/d1bbf52b5c3ab14e456af076a8b4b1e7c5c024c1/lib/msal-node/test/crypto/CryptoProvider.spec.ts#L49) in msal-node. I'm surprised we don't have these in msal-browser as well. Would be good to get these added here too

**Review by tnorling — APPROVED** [review]
Looks great, thanks for addressing this quickly!

File: lib/msal-browser/test/encode/Base64Decode.spec.ts (hunk 2 above)
Comment (sameerag) [inline lib/msal-browser/test/encode/Base64Decode.spec.ts:99]: Do we already have corresponding encode test?

**Review by sameerag — APPROVED** [review]

File: lib/msal-browser/src/encode/Base64Decode.ts (hunk 1 above)
Comment (lalimasharda) [reply lib/msal-browser/src/encode/Base64Decode.ts]: Done

File: lib/msal-browser/test/encode/Base64Decode.spec.ts (hunk 2 above)
Comment (lalimasharda) [reply lib/msal-browser/test/encode/Base64Decode.spec.ts:99]: Added one more case in browser localization tests

Comment (lalimasharda) [reply lib/msal-browser/test/encode/Base64Decode.spec.ts:99]: Added a test for it in the encoding spec.

**Pull request merged**